app/src/middleware/carboneCopy.js

```diff
@@ -0,0 +1,251 @@
+const config = require('config');
+const log = require('npmlog');
+const moment = require('moment');
+const morgan = require('morgan');
+const path = require('path');
+
+const keycloak = require('../components/keycloak');
+
+const clientId = config.get('keycloak.clientId');
+
+const operations = Object.freeze([
+  {method: 'POST', regex: /\/template$/, name: 'UPLOAD_TEMPLATE', isGenerator: false},
+  {method: 'POST', regex: /\/template\/\w+\/render/g, name: 'GENERATE_FROM_TEMPLATE', isGenerator: true},
+  {method: 'POST', regex: /\/template\/render$/, name: 'GENERATE_FROM_UPLOAD', isGenerator: true},
+  {method: 'GET', regex: /\/render/g, name: 'GET_OUTPUT', isGenerator: false},
+  {method: 'GET', regex: /\/template\/(?!render)/g, name: 'GET_TEMPLATE', isGenerator: false}
+]);
+
+const getOperation = (req) => {
+  const result = {isTrackable: false, isGenerator: false, name: null};
+  if (!req) return result;
+
+  operations.forEach(o => {
+    if (!result.operation) {
+      if (req.method === o.method && req.url.match(o.regex)) {
+        result.name = o.name;
+        result.isTrackable = true;
+        result.isGenerator = o.isGenerator;
+      }
+    }
+  });
+
+  return result;
+};
+
+const apiFormat = ':op :azp :ts :status :response-time';
+const generatorApiFormat = `${apiFormat} :outputFileType :contextKeyCount :contentFileType :contentEncodingType :contentSize :res[content-length]`;
+
+const apiTracker = async (req, res, next) => {
+  const operation = req._carboneOp;
+
+  if (operation && operation.isTrackable) {
+    req._ts = moment.utc().valueOf();
+    req._op = operation.name;
+  }
+  next();
+};
+
+const apiTrackerParse = (msg) => {
+  if (msg && msg.trim().length) {
+    const parts = msg.trim().split(' ');
+
+    const ms = Number.parseFloat(parts[4]);
+    const s = ms / 1000;
+    const roundedSeconds = Math.round(s * 10) / 10;
```

Comment:
Could be a util function handling time stuff (if this is needed anywhere else or general time stuff is needed)

Reply:
Didn't do this, because when v1 is gone, there are basically no other classes - middleware and app...